components/StorageFileList.vue

```diff
@@ -1,27 +1,60 @@
 <script setup lang="ts">
 import { FileObject } from "@supabase/storage-js";
+import { FilterParams } from "@/types/FilterParams";
+import { QueryParams } from "@/types/QueryParams";
 
 const sortColumn = useSortColumn();
 const sortOrder = useSortOrder();
 const layoutType = useLayoutType();
 const selectedFiles = useSelectedFiles();
 
+const filters = ref<FilterParams>({
+  name: null,
+  types: null,
+  sizeMin: null,
+  sizeMax: null,
+  dates: null,
+});
+
 const computedClass = computed(() => {
   return layoutType.value === "grid"
     ? "files__list--grid"
     : "files__list--list";
 });
 
+const queryParameters = computed(() => {
+  const queryObject: QueryParams = {
+    key: sortColumn.value,
+    order: sortOrder.value,
+  };
+  if (filters.value.name) {
+    queryObject.name = filters.value.name;
+  }
+  if (filters.value.types?.length) {
+    queryObject.types = filters.value.types.join(",");
+  }
+  if (filters.value.sizeMin) {
+    queryObject.minSize = filters.value.sizeMin;
+  }
+  if (filters.value.sizeMax && filters.value.sizeMax < 5) {
+    queryObject.maxSize = filters.value.sizeMax;
+  }
+  if (filters.value.dates?.length) {
+    queryObject.dates = filters.value.dates.join(",");
+  }
+  return queryObject;
+});
+
 const {
   data: fileList,
   refresh,
   error: fileListError,
 } = await useFetch<FileObject[]>(`/api/files`, {
-  query: { key: sortColumn, order: sortOrder },
+  query: queryParameters,
 });
 
-const updateList = () => {
-  refresh();
+const updateList = async () => {
```

Comment:
is it needed for this function to be `async`?

Reply:
Right, thank you, probably it left from my client side filtering :)